### PR TITLE
Feature/repo page

### LINKED ISSUE #121: [SupportPlatform-Service] Implement repository contributed file types metrics

### LINKED ISSUE #122: [SupportPlatform-Service] Implement repository issue contributions metrics

### LINKED ISSUE #123: [SupportPlatform-Service] Implement metrics of number and percentage of repository files

### LINKED ISSUE #124: [Supportplatform-Web] Implement repository interface

### LINKED ISSUE #125: [Supportplatform-Web] Implement metric interface of number and percentage of repository commits

### LINKED ISSUE #126: [Supportplatform-Web] Implement metric interface of number of lines changed from repository

### LINKED ISSUE #127: [Supportplatform-Web] Implement repository contributed file types metric interface

### LINKED ISSUE #128: [Supportplatform-Web] Implement repository issue contributions metric interface

### LINKED ISSUE #129: [Supportplatform-Web] Implement metric interface of number and percentage of repository files

### LINKED ISSUE #130: [SupportPlatform-Service] Implement commit description quality metric

### LINKED ISSUE #131: [Supportplatform-Web] Implement commit description quality metric interface

### PR DESCRIPTION
**Pull request by guizombas**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

# Type of change
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance improvement (change that optimizes software performance)
- [ ] Code refactoring (restructuring of the source code without changing external functionality)
- [ ] Usability enhancement (change that improves the user experience)
- [ ] Dependency update (updating libraries or external components)
- [ ] Security fix (change that addresses a security vulnerability)
- [ ] Documentation (addition or update of software documentation)
- [ ] Removal of deprecated features (elimination of unused functionalities or components)
- [ ] Testing (addition or update of automated tests)
- [ ] Others (specify the type of change)

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Browser: (e.g., Google Chrome, Mozilla Firefox, Safari)
* Device: (e.g., desktop, laptop, tablet, smartphone)
* Screen Resolution: 
* Package Manager: (e.g., npm, Yarn)
* Docker Containerization: (e.g., Yes/No)

# Checklist:

## Backend Testing Checklist:
- [ ] Test endpoints using Postman:
  - [ ] Verify if endpoints are responding correctly.
  - [ ] Test different HTTP methods (GET, POST, PUT, DELETE, etc.).
  - [ ] Check returned status codes (200, 400, 500, etc.).
  - [ ] Send requests with different parameter sets.
  - [ ] Test authentication and authorization cases if applicable.
  - [ ] Verify the validity and consistency of returned data.
  - [ ] Create a Postman collection with all the necessary requests.
- [ ] Verify if the application functions correctly with Docker Compose.
- [ ] Verify if the application functions correctly without Docker Compose in the local environment.

## Frontend Testing Checklist:
- [ ] Responsiveness:
  - [ ] Test on different devices (desktop, laptop, tablet, smartphone).
  - [ ] Verify appearance and layout on different screen sizes.
  - [ ] Test functionality in both landscape and portrait mode on mobile devices.
- [ ] Browsers:
  - [ ] Test on popular browsers (Google Chrome, Mozilla Firefox, Safari).
  - [ ] Check compatibility and consistent behavior across different browsers.

## Automated Testing:
- [ ] Test Coverage: Evaluate the coverage of automated tests.
- [ ] Unit Tests: Execute unit tests to verify specific functionality.
- [ ] Integration Tests: Perform integration tests to check component interactions.

Closes #121 
Closes #122 
Closes #123 
Closes #124 
Closes #125 
Closes #126 
Closes #127 
Closes #128 
Closes #129 
Closes #130 
Closes #131 
Closes #125

### CODE REVIEW

**Comment by guizombas** [comment]
@Lucas-Angelo todas as alterações discutidas realizadas 👌

File: Codigo/dtos-lib/dto/issueMetrics.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  issuesOpennedCount: number;
+  issuesClosedCount: number;
+}
+
+export interface IssueMetricQueryDTO {
+  /** @isDate startedAt must be a valid date  */
+  startedAt?: Date;
+  /** @isDate endedAt must be a valid date */
+  endedAt?: Date;
+  contributor?: Array<string>;
```
Comment (Lucas-Angelo) [inline Codigo/dtos-lib/dto/issueMetrics.ts:22]: @isArray
@minItems

File: Codigo/dtos-lib/dto/issueMetrics.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  filterWithNoContributor?: boolean;
```
Comment (Lucas-Angelo) [inline Codigo/dtos-lib/dto/issueMetrics.ts:24]: @isBool

File: Codigo/dtos-lib/dto/repositoryMetrics.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface RepositoryMetricQueryDTO {
+    /** @isDate startedAt must be a valid date  */
+    startedAt?: Date,
+    /** @isDate endedAt must be a valid date */
+    endedAt?: Date,
+    branchName?: string,
+    contributor?: Array<string>,
```
Comment (Lucas-Angelo) [inline Codigo/dtos-lib/dto/repositoryMetrics.ts:8]: @isArray
@minItems

File: Codigo/dtos-lib/dto/repositoryMetrics.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    filterWithNoContributor?: boolean
```
Comment (Lucas-Angelo) [inline Codigo/dtos-lib/dto/repositoryMetrics.ts:10]: @isBool

File: Codigo/supportplatform-service/env/.env.development.sample [hunk 5]
```diff
@@ -21,3 +21,7 @@ JWT_SECRET=
 
 # Session configs
 SESSION_SECRET=
+
+# OAUTH redirect url
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/env/.env.development.sample]: Necessário adicionar nos outros samples, test e prod

File: Codigo/supportplatform-service/src/controller/BranchController.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import { Get, Queries, Res, Route, Security, Tags, TsoaResponse } from "tsoa";
+import BranchService from "../service/BranchService";
+import { ErrorResponseDTO, GetAllBranchQueryDTO } from "@gitgrade/dtos";
+import { BranchMapper } from "../mapper/BranchMapper";
+import RepositoryService from "../service/RepositoryService";
+
+@Security("bearer", ["admin"])
+@Route("repository/{repositoryId}/branch")
+@Tags("branch")
+export class BranchController {
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/BranchController.ts]: extends Controller

File: Codigo/supportplatform-service/src/controller/BranchController.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    constructor() {
+        this.branchService = new BranchService();
+        this.repositoryService = new RepositoryService();
+    }
+    @Get("/")
+    async getByRepositoryId(
+        repositoryId: number,
+        @Queries() query: GetAllBranchQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>
+    ) {
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/BranchController.ts:23]: definir this.setStatus(200);

File: Codigo/supportplatform-service/src/controller/BranchController.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+@Security("bearer", ["admin"])
+@Route("repository/{repositoryId}/branch")
+@Tags("branch")
+export class BranchController {
+    private repositoryService: RepositoryService;
+    private branchService: BranchService;
+
+    constructor() {
+        this.branchService = new BranchService();
+        this.repositoryService = new RepositoryService();
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/BranchController.ts]: Na controller de Branch deve haver apenas o service de branch, padrão de backend. Se utilizará alguma coisa do RepositoryService, leva a regra para dentro do BranchService

File: Codigo/supportplatform-service/src/controller/BranchController.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+        const serviceResponse = await this.branchService.getByRepositoryId(
+            repositoryId,
+            {
+                limit: query.limit || 10,
+                page: query.page || 1,
+            }
+        );
+        const mapper = new BranchMapper();
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/BranchController.ts]: Faça o mapper com injeção de dependência no construtor da controller também

File: Codigo/supportplatform-service/src/controller/ContributorController.ts [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+import ContributorMapper from "../mapper/ContributorMapper";
+import { ErrorResponseDTO, GetAllContributorQueryDTO } from "@gitgrade/dtos";
+import RepositoryService from "../service/RepositoryService";
+
+@Security("bearer", ["admin"])
+@Route("repository/{repositoryId}/contributor")
+@Tags("contributor")
+export class ContributorController {
+    private contributorService: ContributorService;
+    private repositoryService: RepositoryService;
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/ContributorController.ts]: Idem, leva a regra para dentro do service.

File: Codigo/supportplatform-service/src/controller/ContributorController.ts [hunk 11]
```diff
@@ -0,0 +1,46 @@
+import { Get, Queries, Res, Route, Security, Tags, TsoaResponse } from "tsoa";
+import ContributorService from "../service/ContributorService";
+import ContributorMapper from "../mapper/ContributorMapper";
+import { ErrorResponseDTO, GetAllContributorQueryDTO } from "@gitgrade/dtos";
+import RepositoryService from "../service/RepositoryService";
+
+@Security("bearer", ["admin"])
+@Route("repository/{repositoryId}/contributor")
+@Tags("contributor")
+export class ContributorController {
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/ContributorController.ts]: Idem, extends controller.

File: Codigo/supportplatform-service/src/controller/ContributorController.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+        this.contributorService = new ContributorService();
+        this.repositoryService = new RepositoryService();
+    }
+
+    @Get("/")
+    async findbyRepositoryId(
+        repositoryId: number,
+        @Queries() query: GetAllContributorQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>
+    ) {
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/ContributorController.ts:24]: Idem, define status.

File: Codigo/supportplatform-service/src/controller/ContributorController.ts [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        const serviceResponse = await this.contributorService.getByRepositoryId(
+            repositoryId,
+            {
+                limit: query.limit || 10,
+                page: query.page || 1,
+            }
+        );
+        const mapper = new ContributorMapper();
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/ContributorController.ts]: Idem, mapper injetado na controller.

File: Codigo/supportplatform-service/src/controller/RepositoryController.ts [hunk 14]
```diff
@@ -50,12 +52,30 @@ export class RepositoryController extends Controller {
...
 
     /**
+     * @param id @isInt id id must be an integer
+     * @param id @minimum id 1 id must be greater than or equal to 1
+     */
+    @Get("/{id}")
+    public async getById(
+        id: number,
+        @Res() notFoundResponse: TsoaResponse<404, { message: string }>
+    ): Promise<RepositoryResponseDTO> {
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryController.ts:66]: Idem, define status.

File: Codigo/supportplatform-service/src/controller/RepositoryController.ts [hunk 15]
```diff
@@ -50,12 +52,30 @@ export class RepositoryController extends Controller {
...
+        const serviceResponse = await this.repositoryService.findById(id);
+
+        if (!serviceResponse) {
+            return notFoundResponse(404, { message: "Repository not found" });
+        }
+        const mapper = new RepositoryMapper();
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryController.ts:72]: Idem, injeta mapper no construtor.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 16]
```diff
@@ -1,58 +1,254 @@
...
     @Get("commit")
     @SuccessResponse("200", "Commit metrics")
     async getCommitMetrics(
         repositoryId: number,
-        @Query() branchName?: string
+        @Queries() query: RepositoryMetricQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>,
+        @Res() unprocessableEntityResponse: TsoaResponse<422, ErrorResponseDTO>
     ): Promise<CommitMetricsDTO> {
-        this.setStatus(200);
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts:35]: Por que tirou?

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 17]
```diff
@@ -1,58 +1,254 @@
...
-                branchName ?? "master"
+                branchName,
+                startedAt,
+                endedAt,
+                query.contributor,
+                query.filterWithNoContributor
             );
+
+        this.setStatus(200);
         return new CommitMetricsMapper().toDto(serviceResponse);
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Idem, injeta o mapper.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 18]
```diff
@@ -1,58 +1,254 @@
...
                 repositoryId,
-                branchName ?? "master"
+                branchName,
+                startedAt,
+                endedAt,
+                query.contributor,
+                query.filterWithNoContributor
             );
+        this.setStatus(200);
         return new FileChangeMetricsMapper().toDto(serviceResponse);
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Idem, injeta o mapper.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 19]
```diff
@@ -1,58 +1,254 @@
...
     @SuccessResponse("200", "Commit metrics")
     async getCommitMetrics(
         repositoryId: number,
-        @Query() branchName?: string
+        @Queries() query: RepositoryMetricQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>,
+        @Res() unprocessableEntityResponse: TsoaResponse<422, ErrorResponseDTO>
     ): Promise<CommitMetricsDTO> {
-        this.setStatus(200);
+        const validateQueryInterval = new QueryIntervalValidator(
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Acredito que poderia injetar esse QueryValidator no construtor também.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 20]
```diff
@@ -1,58 +1,254 @@
...
     @SuccessResponse("200", "File change metrics")
     async getChangesMetrics(
         repositoryId: number,
-        @Query() branchName?: string
+        @Queries() query: RepositoryMetricQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>,
+        @Res() unprocessableEntityResponse: TsoaResponse<422, ErrorResponseDTO>
     ) {
-        this.setStatus(200);
+        const validateQueryInterval = new QueryIntervalValidator(
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Idem, injeta no construtor.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 21]
```diff
@@ -1,58 +1,254 @@
...
     }
+
+    @Get("file-types")
+    async getFileTypesMetrics(
+        repositoryId: number,
+        @Queries() query: RepositoryMetricQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>,
+        @Res() unprocessableEntityResponse: TsoaResponse<422, ErrorResponseDTO>
+    ) {
+        const validateQueryInterval = new QueryIntervalValidator(
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Idem, injeta no construtor.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 22]
```diff
@@ -1,58 +1,254 @@
...
+                startedAt,
+                endedAt,
+                query.contributor,
+                query.filterWithNoContributor
             );
+        this.setStatus(200);
         return new FileChangeMetricsMapper().toDto(serviceResponse);
     }
+
+    @Get("file-types")
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts:138]: esse não tem annotations de @isInt e @minimum não?

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 23]
```diff
@@ -1,58 +1,254 @@
...
+        const repository = await this.repositoryService.findById(repositoryId);
+
+        if (!repository) {
+            return notFoundResponse(404, {
+                message: "Repository not found",
+            });
+        }
+
+        const branchName =
+            query.branchName ?? repository.defaultBranch ?? "master";
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Isso aqui vai dar M, ao template novo dos TIs novos já utilizam main. Acho que alterei defaultBranch pra nullable sem querer uma época, mas ele é not null no GitHub.

File: Codigo/supportplatform-service/src/controller/BranchController.ts [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+    @Get("/")
+    async getByRepositoryId(
+        repositoryId: number,
+        @Queries() query: GetAllBranchQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>
+    ) {
+        const repository = await this.repositoryService.findById(repositoryId);
+
+        if (!repository) {
+            return notFoundResponse(404, {
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/BranchController.ts]: Grande parte do código já utiliza AppError com 404 para tratar resources not found, vamos ficar despadronizados.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 25]
```diff
@@ -1,58 +1,254 @@
...
+     * @param repositoryId @minimum repositoryId 1 repositoryId must be greater than or equal to 1
+     */
     @Get("commit")
     @SuccessResponse("200", "Commit metrics")
     async getCommitMetrics(
         repositoryId: number,
-        @Query() branchName?: string
+        @Queries() query: RepositoryMetricQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>,
+        @Res() unprocessableEntityResponse: TsoaResponse<422, ErrorResponseDTO>
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Tenho minhas duvidas de despadronizar o código não utilizando essa regra no service com AppError.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 26]
```diff
@@ -1,58 +1,254 @@
...
+    }
+
+    @Get("issues")
+    async getIssuesMetrics(
+        repositoryId: number,
+        @Queries() query: IssueMetricQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>,
+        @Res() unprocessableEntityResponse: TsoaResponse<422, ErrorResponseDTO>
+    ) {
+        const validateQueryInterval = new QueryIntervalValidator(
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Idem, injeta no construtor.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 27]
```diff
@@ -1,58 +1,254 @@
...
+     * @param repositoryId @minimum repositoryId 1 repositoryId must be greater than or equal to 1
+     */
+    @Get("commit-quality")
+    async getCommitQualityMetrics(
+        repositoryId: number,
+        @Queries() query: RepositoryMetricQueryDTO,
+        @Res() notFoundResponse: TsoaResponse<404, ErrorResponseDTO>,
+        @Res() unprocessableEntityResponse: TsoaResponse<422, ErrorResponseDTO>
+    ): Promise<CommitQualityMetricsDTO> {
+        const validateQueryInterval = new QueryIntervalValidator(
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Idem, injeta no construtor.

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts [hunk 28]
```diff
@@ -1,58 +1,254 @@
...
+        const serviceResponse =
+            await this.commitService.getCommitQualityGroupedByContributor(
+                repositoryId,
+                branchName,
+                startedAt,
+                endedAt,
+                query.contributor,
+                query.filterWithNoContributor
+            );
+        return new CommitQualityMetricsMapper().toDto(serviceResponse);
```
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: Idem, injeta no construtor.

File: Codigo/supportplatform-service/src/model/Commit.ts
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/model/Commit.ts:1]: Atualiza no JobScheduler também.

File: Codigo/supportplatform-service/src/model/Contributor.ts
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/model/Contributor.ts:1]: Atualiza no JobScheduler também.

File: Codigo/supportplatform-service/src/model/File.ts
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/model/File.ts:1]: Atualiza no JobScheduler também.

File: Codigo/supportplatform-service/src/model/Issue.ts
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/src/model/Issue.ts:1]: Atualiza no JobScheduler também.

File: Codigo/supportplatform-service/tsconfig.3eacndjscjy.json
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/tsconfig.3eacndjscjy.json:1]: parece coisa que não devia ter subido

File: Codigo/supportplatform-service/tsconfig.i0sagct6dl.json
Comment (Lucas-Angelo) [inline Codigo/supportplatform-service/tsconfig.i0sagct6dl.json:1]: parece coisa que não devia ter subido, 2

**Review by Lucas-Angelo — DISMISSED** [review]

File: Codigo/dtos-lib/dto/issueMetrics.ts (hunk 1 above)
Comment (guizombas) [reply Codigo/dtos-lib/dto/issueMetrics.ts:22]: minItems não é necessário pois é um filtro (opcional), isArray vou adicionar

File: Codigo/supportplatform-service/env/.env.development.sample (hunk 5 above)
Comment (guizombas) [reply Codigo/supportplatform-service/env/.env.development.sample]: feito na branch de login-logout

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts (hunk 22 above)
Comment (guizombas) [reply Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts:138]: comentário tá no lugar certo?

Comment (guizombas) [reply Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts:138]: tá certo, entendi agora

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts (hunk 23 above)
Comment (guizombas) [reply Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: beleza, se me garante que ele vem algo eu passo o bang operator (! sim esse é o nome desse trem kk) pra dizer que ele não vai ser null

File: Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts (hunk 25 above)
Comment (guizombas) [reply Codigo/supportplatform-service/src/controller/RepositoryMetricsController.ts]: farei a alteração pra padronizar com AppError

**Pull request merged**